clicache/integration-test2/FunctionExecutionTest.cs

```diff
@@ -0,0 +1,151 @@
+                    .execute(), 0);
+
+                var cacheFactory = new CacheFactory();
+                var cacheOne = cacheFactory.Create();
+                cacheOne.GetPoolFactory()
+                    .AddLocator(cluster.Gfsh.LocatorBindAddress, cluster.Gfsh.LocatorPort)
+                    .Create("pool");
+                var regionFactory = cacheOne.CreateRegionFactory(RegionShortcut.PROXY)
+                    .SetPoolName("pool");
+                var region = regionFactory.Create<object, object>("testRegion1");
+
+                for (int i = 0; i < 230; i++)
+                {
+                  region["KEY--" + i] = "VALUE--" + i;
+                }
+
+                object args = true;
+              
+
+                Object[] oddKeyFilter = new Object[17];
+                int j = 0;
+                for (int i = 0; i < 34; i++)
+                {
+                  if (i % 2 == 0) continue;
+                  oddKeyFilter[j] = "KEY--" + i;
+                  j++;
+                }
+                Apache.Geode.Client.Execution<object> exc = Client.FunctionService<object>.OnRegion<object, object>(region);
+                Client.IResultCollector<object> rc = exc.WithArgs<object>(args).WithFilter<object>(oddKeyFilter).Execute("MultiGetFunction");
+                ICollection<object> executeFunctionResult = rc.GetResult();
+                List<object> resultList = new List<object>();
+
+                foreach (List<object> item in executeFunctionResult)
+                {
+                  foreach (object item2 in item)
+                  {
+                    resultList.Add(item2);
+                  }
+                }
+                Assert.True(resultList.Count == expectedFilteredCount, "result count check failed");
+
+            }
+        }
+
+    [Fact]
+    public void MultiGetIFunctionExecutionWithArgs()
+    {
+      int expectedResultCount = 17;
+      using (var cluster = new Cluster(CreateTestCaseDirectoryName(), 1, 1))
+      {
+        Assert.Equal(cluster.Start(), true);
+        Assert.Equal(cluster.Gfsh.deploy()
+            .withJar(Config.JavaobjectJarPath)
+            .execute(), 0);
+        Assert.Equal(cluster.Gfsh
+            .create()
+            .region()
+            .withName("partition_region")
+            .withType("PARTITION")
+            .execute(), 0);
+
+        var cacheFactory = new CacheFactory();
+        var cacheOne = cacheFactory.Create();
+        cacheOne.GetPoolFactory()
+            .AddLocator(cluster.Gfsh.LocatorBindAddress, cluster.Gfsh.LocatorPort)
+            .Create("pool");
+        var regionFactory = cacheOne.CreateRegionFactory(RegionShortcut.PROXY)
+            .SetPoolName("pool");
+        var region = regionFactory.Create<object, object>("partition_region");
+
+        for (int i = 0; i < 230; i++)
+        {
+          region["KEY--" + i] = "VALUE--" + i;
+        }
+
+        object args = true;
+        Object[] oddKeyFilter = new Object[17];
```

Comment:
I feel that we're adding unnecessary complexity. Why are we building oddKeyFilter, if the only thing we're going to do with it is add each element to oddKeyArgs? Why not just build the latter and throw out the former?